### PR TITLE
fix: add tests to maximize the test coverage

### LINKED ISSUE #46: Ensure test coverage of views and models is sufficiently high
After all models and views have been migrated to v2, it is imperative that we keep a good test coverage of our code base. We should not necessarily aim for 100%, but any blob of code or if/else condition from the view or model modules that is not covered is suspect. If we do find such tests, then we should make sure that we have adequately migrated all the tests from v1. If somes pieces of code were not covered in v1, then we should create new tests in v2.

### PR DESCRIPTION
**Pull request by Ali-Salman29**
The pull request contains tests to maximize the test coverage.

- We have increased the coverage from 84 to 87, and we plan to improve it further in future PRs.
- The test coverage should be expanded for utilities and elastic search classes. To keep track of this, we have created issue #96.

close #46

### CODE REVIEW

File: forum/views/comments.py [hunk 1]
```diff
@@ -35,10 +35,20 @@ def validate_comments_request_data(data: dict[str, Any]) -> None:
...
 
 
+def get_thread_id(parent_comment_id: str) -> str:
+    """
+    The thread Id from the parent comment.
+    """
+    parent_comment = Comment().get(parent_comment_id)
+    if parent_comment:
+        return parent_comment["comment_thread_id"]
+    raise ValueError("Comment doesn't have the thread.")
```
Comment (Faraz32123) [inline forum/views/comments.py:45]: no need to raise a value_error as comment_thread_id will always exists, it's a required field.

File: forum/views/subscriptions.py [hunk 2]
```diff
@@ -46,9 +46,13 @@ def post(self, request: Request, user_id: str) -> Response:
...
-        thread = None
-        if request_data.get("source_id"):
-            thread = CommentThread().get(request_data["source_id"])
+        source_id = request_data.get("source_id")
+        if not source_id:
+            return Response(
+                {"error": "source_id is required"},
+                status=status.HTTP_400_BAD_REQUEST,
+            )
+        thread = CommentThread().get(request_data["source_id"])
```
Comment (Faraz32123) [inline forum/views/subscriptions.py:55]: `thread = CommentThread().get(source_id)`

File: tests/test_models/test_threads.py [hunk 3]
```diff
@@ -3,9 +3,34 @@
...
+        )
+
+    with pytest.raises(ValueError, match="Invalid context"):
+        CommentThread().insert(
+            title="Test title",
+            body="Test body",
+            course_id="course_123",
+            commentable_id="commentable_123",
+            author_id="author_123",
+            context="invalid_context",  # Invalid context
```
Comment (Faraz32123) [inline tests/test_models/test_threads.py:30]: we can remove these comments,
# Invalid context, # Invalid thread_type

**Review by taimoor-ahmed-1 — APPROVED** [review]
LGTM

**Review by Faraz32123 — APPROVED** [review]

**Pull request merged**